editor/scene_tree_dock.cpp

```diff
@@ -2263,31 +2331,114 @@ void SceneTreeDock::_shader_creation_closed() {
 	shader_create_dialog->disconnect("canceled", callable_mp(this, &SceneTreeDock::_shader_creation_closed));
 }
 
+void SceneTreeDock::_disable_all_editable_children_from_selection() {
+	editable_instance_enable_or_disable_dialog->hide();
+	dialog_option_enable_all = false;
+
+	// Fire confirmation dialog.
+	editable_instance_remove_dialog->set_text(TTR("Disabling \"Editable Children\" will cause all properties of those nodes to be reverted to their default."));
+	editable_instance_remove_dialog->popup_centered();
+}
+
+void SceneTreeDock::_enable_all_editable_children_from_selection() {
+	dialog_option_enable_all = true;
+	_toggle_editable_children_from_selection();
+}
+
 void SceneTreeDock::_toggle_editable_children_from_selection() {
 	List<Node *> selection = editor_selection->get_selected_node_list();
-	List<Node *>::Element *e = selection.front();
 
-	if (e) {
-		_toggle_editable_children(e->get());
+	if (selection.size() == 1) {
+		List<Node *>::Element *e = selection.front();
+		if (e) {
+			_toggle_editable_children(e->get());
+		}
+	} else {
+		for (Node *node : selection) {
+			if (node) {
```

Comment:
```suggestion
			if (!node) {
				continue;
			}
```
Less indentation.

I think you could also remove the upper condition; range loop for a single element does not really make difference.

Reply:
I remember I added the `if (node)` checks when I was just getting started and saw it being used elsewhere, but I have since wondered if they're necessary at all?[^1] I grep'd through the codebase and only really saw such checks being used in situations where it was clearly sensible to do so, while my use of it seems overly paranoid.

[^1]: I figured maybe it guards against a node being freed in the meantime. I removed the checks and tested with a tool node that can free itself while the rmb menu is open; because doing so also removes it from the scene tree dock and therefore the selection, it causes no issues. Seems the selection and the scene would have to diverge in the middle of the operation, or the selection array be fouled up some other way, both of which seem improbable. Though I think I now speculate on things outside my understanding...

Reply:
Re: removing the condition for single-node selection: it probably wouldn't save much space vertically as is (though would further reduce indentation). We would still need to catch the single-node case in order to set `dialog_option_enable_all = !editable`, before falling through to the for-loop. Like this:

```
	if (selection.size() == 1) {
		List<Node *>::Element *e = selection.front();
		if (e) {
			dialog_option_enable_all = !EditorNode::get_singleton()->get_edited_scene()->is_editable_instance(e->get());
		}
	}
	for (Node *node : selection) {
		if (!node) {
			continue;
		}
		bool editable = !EditorNode::get_singleton()->get_edited_scene()->is_editable_instance(node);

		if (editable == dialog_option_enable_all) {
			_toggle_editable_children(node);
		}
	}
```

Maybe I should just add a `_set_editable_children(node, editable)` method? Something roughly like this:

```
	for (Node *node : selection) {
		if (!node) {
			continue;
		}
		_set_editable_children(node, dialog_option_enable_all);
	}

...

void SceneTreeDock::_set_editable_children(Node *p_node, bool p_editable) {
	if (p_node) {
		bool editable = EditorNode::get_singleton()->get_edited_scene()->is_editable_instance(p_node);
		if (!editable == p_editable) {
			// set here or call _toggle_editable_children
			// (if set here, maybe make _toggle_editable_children call here instead to avoid duplication)
		}
	}
}
```

Reply:
So `dialog_option_enable_all` is not initialized for single node? Then I guess it's not worth to change it.